keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1806 @@
+correctly, it must be enabled in all parts of the cluster (scheduler, API server,
+kubelet). 
+
+When the feature gate is disabled on control plane, but enabled on kubelet,
+users will not be able to schedule Pods with the new `resources` field in the
+`spec`.
+
+When the feature gate is enabled on control plane, but disabled on kubelet, users
+will be able to create the Pods with `resources` field, but kubelet will reject
+those pods.
+
+Therefore, all cluster nodes (controlplane and worker nodes), must be upgraded
+before the user can deploy pods with the new `resources` field in the `spec`.
+
+For users wanting to use this feature, it is user's responsibility to use the
+correct version of kube-scheduler, kube-apiserver and kubelet otherwise the
+feature will not work as expected.
+
+## Production Readiness Review Questionnaire
+
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: PodLevelResources
+  - Components depending on the feature gate: kubelet, kube-apiserver, kube-scheduler
+  - Will enabling / disabling the feature require downtime of the control
+    plane? No. Once the feature is disabled, the control plane components will
+    stop using the pod-level resources from the spec.
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node? No. Once the feature is disabled, the kubelet will stop using the pod
+    level resources from the spec.
+
+###### Does enabling the feature change any default behavior?
+
+No. This feature is guarded by a feature gate, and requires setting Pod level `resource` stanza explicitly.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Any new pods created after disabling the feature will not utilize the
+pod-level resources feature. Disabling the feature will not affect existing pods
+-- they will continue to function. 
+
+However, tfor pods created with pod-level resources, there will be a mismatch
+between the resource calculations and the actual usage. To fix thie issue, users
+may delete the running pods and create new ones.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+Reenabling the feature will not affect existing pods. Once reenabled, any new
+pods can be created with pod-level resources to use the feature.
```

Comment:
Can we say any more about what happens to a pod that used pod-level resources before the the feature was disabled?  I'm not quite sure what "will not effect'  implies here.  Would such pods have the pod-level resource fields set but no honored?  If so should we recommend the pods be recreated again after reenabling the feature?